src/core/prometheus.cc

```diff
@@ -800,19 +800,19 @@ class metrics_handler : public httpd::handler_base  {
 
     future<std::unique_ptr<http::reply>> handle(const sstring& path,
         std::unique_ptr<http::request> req, std::unique_ptr<http::reply> rep) override {
-        auto text = is_accept_text(req->get_header("Accept"));
+        auto text_format = !_ctx.allow_protobuf || !is_accept_protobuf(req->get_header("Accept"));
         sstring metric_family_name = req->get_query_param("__name__");
         bool prefix = trim_asterisk(metric_family_name);
         bool show_help = req->get_query_param("__help__") != "false";
         std::function<bool(const mi::labels_type&)> filter = make_filter(*req);
 
-        rep->write_body((text || !_ctx.allow_protobuf) ? "txt" : "proto", [this, text, metric_family_name, prefix, show_help, filter] (output_stream<char>&& s) {
+        rep->write_body((text_format) ? "txt" : "proto", [this, text_format, metric_family_name, prefix, show_help, filter] (output_stream<char>&& s) {
             return do_with(metrics_families_per_shard(), output_stream<char>(std::move(s)),
-                    [this, text, prefix, &metric_family_name, show_help, filter] (metrics_families_per_shard& families, output_stream<char>& s) mutable {
-                return get_map_value(families).then([&s, &families, this, text, prefix, &metric_family_name, show_help, filter]() mutable {
+                    [this, text_format, prefix, &metric_family_name, show_help, filter] (metrics_families_per_shard& families, output_stream<char>& s) mutable {
+                return get_map_value(families).then([&s, &families, this, text_format, prefix, &metric_family_name, show_help, filter]() mutable {
                     return do_with(get_range(families, metric_family_name, prefix),
-                            [&s, this, text, show_help, filter](metric_family_range& m) {
-                        return (text) ? write_text_representation(s, _ctx, m, show_help, filter) :
+                            [&s, this, text_format, show_help, filter](metric_family_range& m) {
+                        return (text_format) ? write_text_representation(s, _ctx, m, show_help, filter) :
```

Comment:
i'd suggest include this very change in the previous one. as it addresses a bug in it.

Reply:
I can squas the changes, I tried to keep the original as close as possible to make it easier for maintainer to understand what was changed.

Reply:
the bug was introduced in the commit of "[Prometheus: Disable Prometheus protobuf with a configuration](https://github.com/scylladb/seastar/pull/1818/commits/f3d8620cdfe5798b643c323a135cf109142a65f5)", so you are fixing the issue in the very same PR which introduces it.